### PR TITLE
Send null when nullable

### PR DESCRIPTION
**Pull request by Paultagoras**
## Summary
* Send null when we have null primitives and a nullable column

Closes https://github.com/ClickHouse/clickhouse-java/issues/2122
## Checklist
Delete items not relevant to your PR:
- [x] Closes #<issue ref>
- [x] Unit and integration tests covering the common scenarios were added
- [x] A human-readable description of the changes was provided to include in CHANGELOG
- [ ] For significant changes, documentation in https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

### CODE REVIEW

**Comment by chernser** [comment]
@Paultagoras would you please add a test for JDBC to check that we are getting null? I remember it has separate class or so. 
Also no `Array(Nullable(Int32))` test case.

File: client-v2/src/main/java/com/clickhouse/client/api/data_formats/internal/SerializerUtils.java [hunk 1]
```diff
@@ -462,6 +462,12 @@ private static void serializeMapData(OutputStream stream, Object value, ClickHou
     }
 
     private static void serializePrimitiveData(OutputStream stream, Object value, ClickHouseColumn column) throws IOException {
+        //Handle null values
+        if (value == null && column.isNullable()) {//Only nullable columns can have null values
```
Comment (mzitnik) [inline client-v2/src/main/java/com/clickhouse/client/api/data_formats/internal/SerializerUtils.java:466]: If it null and not nullable should we throw an exception like data is not valid

**Review by mzitnik — APPROVED** [review]
